compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -990,17 +990,26 @@ object SymDenotations {
     def isInlineMethod(implicit ctx: Context): Boolean =
       isAllOf(InlineMethod, butNot = Accessor)
 
-    def isRetainedInlineMethod(using Context): Boolean =
-      isAllOf(InlineMethod, butNot = AccessorOrDeferred)
+    /** Does this method or field need to be retained at runtime */
+    def isRetainedInline(using Context): Boolean =
+      isAllOf(Inline, butNot = Deferred)
       && allOverriddenSymbols.exists(!_.is(Inline))
 
+    /** Does this method need to be retained at runtime */
+    def isRetainedInlineMethod(using Context): Boolean =
+      is(Method, butNot = Accessor) && isRetainedInline
+
     /** Is this a Scala 2 macro */
     final def isScala2Macro(implicit ctx: Context): Boolean = is(Macro) && symbol.owner.is(Scala2x)
 
     /** An erased value or an inline method.
      */
     def isEffectivelyErased(implicit ctx: Context): Boolean =
-      is(Erased) || isInlineMethod && !isRetainedInlineMethod
+      is(Erased) || is(Inline) && !isRetainedInline && !hasAnnotation(defn.ScalaStaticAnnot)
```

Comment:
What we need the test `!hasAnnotation(defn.ScalaStaticAnnot)`?

Reply:
The `@static` annotation is supposed to guarantee that the field is emitted in the bytecode with the original signature. Hence we cannot remove the definition. There was one bytecode test with a `final val` that failed without this test.